dashboard/app/views/layouts/_footer.html.haml

```diff
@@ -37,3 +37,18 @@
           %br/
           %br/
           !="&copy; Code.org, #{Time.now.year}"
+
+- if DCDO.get('hoc_secret', nil)
+  // Set secret
+  - # If enabled, set a client-side secret that allows admins to access the HOC 2015 levels.
+  - # This is a temporary measure that will be removed shortly.
+  - if current_user && current_user.admin?
+    :javascript
+      //<![CDATA[
+      localStorage.setItem('hoc_secret', '#{DCDO.get("hoc_secret", nil)}');
+      //]]>
+  - else
+    :javascript
+      //<![CDATA[
+      localStorage.removeItem('hoc_secret');
```

Comment:
The cached level pages use the small_footer rather than the footer so this happens to work.

I agree, it's very fragile! I'll add a conditional so that we don't insert this logic on cached pages.